app/src/main/java/com/duckduckgo/app/browser/duckplayer/DuckPlayerJSHelper.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.duckplayer
+
+import com.duckduckgo.duckplayer.api.DuckPlayer
+import com.duckduckgo.js.messaging.api.JsCallbackData
+import javax.inject.Inject
+import org.json.JSONObject
+
+class DuckPlayerJSHelper @Inject constructor(
```

Comment:
Looks like this should be inside the duck player module?

Reply:
Discussed on MM but leaving a summary here for clarity:

I didn't put the JS helper in Duck Player because I wanted to decouple the Duck Player module as much as possible from the fact that all this uses JS behind the scenes. However, all the functionality/logic is still in the duck player module, this only formats to/from JSON.

Also, I can't move the JS Helper to the Duck Player module because of an "Invalid dependency: :duckplayer-api -> :js-messaging-api", which I need to include JSCallbackData as a return in the interface. I could return a JSONObject, but then I still have to keep putting code in the app and honestly, I don't think it's worth it.